backend/djangoindia/db/admin.py

```diff
@@ -53,6 +57,38 @@ def get_urls(self):
         ]
         return custom_urls + urls
 
+    @transaction.atomic
+    def save_model(self, request, obj, form, change):
+        # This is a new registration
+        if not change:  
+            if obj.event.seats_left > 0:
+                obj.event.seats_left -= 1
+                obj.event.save()
+            else:
+                raise ValidationError("No seats left for this event.")
+        super().save_model(request, obj, form, change)
+
+    @transaction.atomic
+    def delete_model(self, request, obj):

```

Comment:
Function names need to improve. I understand that you wish to achieve this with the models, but transaction functions should refer to the actual object they relate to. 

Examples:

```
def new_registration_atomic

def slot_booking_atomic
```

Reply:
these functions are already present in admin.Modeladmin, I am just overriding.